### PR TITLE
Fix all rubocop offenses #192

### LINKED ISSUE #192: Fix all Rubocop warnings
When I execute the linter on my machine using `$ rubocop`, I get a lot of old warnings and error messages. They don't help me with identifying the new warnings that I introduced on my branch before I push to GitHub and get a 'failed' notification from CodeClimate.

### PR DESCRIPTION
**Pull request by jkbe**
Closes #192.
I used the autocorrect option in RubyMine (probably based on `rubocop -a`) to fix all existing offenses in our project (based on the current Rubocop config file). I also turned on checks for the Ruby 1.9 hash syntax which discourages the use of `=>` in hashes where it's not needed (we agreed on this in today's meeting of Scrum Masters). See [documentation for Style/HashSyntax](http://www.rubydoc.info/gems/rubocop/0.51.0/RuboCop/Cop/Style/HashSyntax) (we use `ruby19`).

### CODE REVIEW

File: spec/views/events/edit.html.erb_spec.rb [hunk 1]
```diff
@@ -6,23 +6,23 @@
...
-      :description => "MyText",
-      :owner => FactoryBot.build(:user),
-      :game_mode => "MyString",
-      :max_teams => 20,
-      :player_type => Event.player_types[Event.player_types.keys.sample],
-      :discipline => "MyString",
-      :deadline => Date.new(2017,11,16),
-      :startdate => Date.new(2017,12,01),
-      :enddate => Date.new(2017,12,05)
+                              :name => "MyString",
```
Comment (fpottbaecker) [inline spec/views/events/edit.html.erb_spec.rb]: While we're at it: Can we turn on [`Style/HashSyntax`](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/style/hash_syntax.rb), which forces the new hash syntax on symbol key only hashes?

**Review by fpottbaecker — CHANGES_REQUESTED** [review]

File: spec/views/events/edit.html.erb_spec.rb (hunk 1 above)
Comment (jkbe) [reply spec/views/events/edit.html.erb_spec.rb]: Yes, you're right. I'll do it.

**Review by lheilmann — APPROVED** [review]

**Comment by fpottbaecker** [comment]
And additionally, there is [`Style/StringLiterals`](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/style/string_literals.rb) set to `single_quotes` (by default) which requires a more consistent quoting style (warning increases offenses by about 600. Do we want that too?

File: spec/views/events/edit.html.erb_spec.rb (hunk 1 above)
Comment (jkbe) [reply spec/views/events/edit.html.erb_spec.rb]: @Pottiman it looks to me like `ruby19_no_mixed_keys` from http://www.rubydoc.info/gems/rubocop/0.45.0/RuboCop/Cop/Style/HashSyntax would fit us best. Did you happen to see another option which discourages use of symbols as keys (where hash rockets are allowed again)?

**Comment by jkbe** [comment]
A call for consistent quoting style hasn't come up yet so I wouldn't do that right now. However, we can discuss that in the next Scrum Master Meeting.

File: spec/views/events/edit.html.erb_spec.rb (hunk 1 above)
Comment (fpottbaecker) [reply spec/views/events/edit.html.erb_spec.rb]: I think the default (`ruby19`) allows rockets in non symbol key hashes, eg
```ruby
a = {
  :a => 1,
  'b' => 2
}
```
is legal.
Do you want to explicitly prohibit this?:
```ruby
a = {
  a: 1,
  'b' => 2
}
```

**Comment by fpottbaecker** [comment]
And finally, you can enable all Rails checks:
```yaml
Rails:
  Enabled: true
```
I do not know whether this is default already.

**Comment by jkbe** [comment]
@Pottiman do you know what the Rails Cop does? When enabled, Rubocop doesn't find any new offenses.

**Comment by fpottbaecker** [comment]
It should enable the [rails cop department](https://github.com/bbatsov/rubocop/tree/master/lib/rubocop/cop/rails), just to be sure.

Maybe it already was on...

File: .rubocop.yml [hunk 2]
```diff
@@ -51,7 +51,8 @@ Layout/FirstParameterIndentation:
 
 # Use Ruby >= 1.9 syntax for hashes. Prefer { a: :b } over { :a => :b }.
 Style/HashSyntax:
-  Enabled: false
+  Enabled: true
+  EnforcedStyle: ruby19
```
Comment (fpottbaecker) [inline .rubocop.yml:55]: This is default, do we need to explicitly state it?

File: spec/views/events/edit.html.erb_spec.rb (hunk 1 above)
Comment (jkbe) [reply spec/views/events/edit.html.erb_spec.rb]: As discussed, let's not do this.

File: .rubocop.yml (hunk 2 above)
Comment (jkbe) [reply .rubocop.yml:55]: It's the style in the config file so far so I would just leave it like this.

**Comment by jkbe** [comment]
@Pottiman about the Rails Cop: Let's leave this out for now as there were no new offenses through this and we don't really know what it does.
This PR's main purpose is to fix all offenses that we had in our project with the existing Rubocop config file and to introduce the new HashSyntax rule that we agreed on in the meeting of Scrum Masters.

**Review by fpottbaecker — APPROVED** [review]

**Pull request merged**